net.bobbo.player-controller/player_scripts/crouch.gd

```diff
@@ -33,4 +40,37 @@ func player_physics_process(delta: float):
 	var scale_difference = new_scale - previous_scale
 	
 	player.velocity.y += scale_difference * 0.5
-	player.height = new_scale
+	
+	# Make it so that the crouch can only go as low as regularly crouching and only high enough such that the player doesn't get stuck if an object is detected above them
+	player.height = clamp(new_scale, _max_player_height / 2, _height_above)
+	
+	# Get the 3D physics space from a Node3D like the player_pivot
+	var space_state = player.model_pivot.get_world_3d().direct_space_state
+	
+	# Calculate origin and ending of raycast
+	var origin = player.position
+	var end = player.position + Vector3(0, _max_player_height / 2, 0)
+	
+	# Create the raycast
+	var query = PhysicsRayQueryParameters3D.create(origin, end)
+	
+	# Exclude the player from raycast calculations and make the ray collide with areas
+	query.exclude = [player]
+	query.collide_with_areas = true
+	
+	# Cast the raycast
+	var result = space_state.intersect_ray(query)
```

Comment:
Instead of `result`, perhaps something like `ceiling_collision`?

Reply:
For some reason I didn't even think of giving that a better name, so we're definitely gonna go with something like that